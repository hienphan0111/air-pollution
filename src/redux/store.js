import { configureStore } from '@reduxjs/toolkit';
import countriesReducer from './countries';
import citiesReducer from './cities';
// import cityReducer from '@/routes/city-detail';
import cityReducer from './city';

const store = configureStore({
  reducer: {
    countries: countriesReducer,
    cities: citiesReducer,
    cityInfo: cityReducer,
  },
});

export default store;
