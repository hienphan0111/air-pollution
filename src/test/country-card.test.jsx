import React from 'react';
import TestRenderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect.js';
import { Provider } from 'react-redux';
import store from '../redux/store';
import CountryCard from '../components/country-card.component';
import axios from 'axios';
import { BrowserRouter, MemoryRouter } from 'react-router-dom';

jest.mock('axios', () => ({
  post: jest.fn(),
  get: jest.fn(),
}));

describe('Render the country lists', () => {
  it('Renders the country correctly', () => {
    const country = {
      id: 2,
      name: 'Poland',
      flag: 'https://flagcdn.com/pl.svg',
      region: 'Asia',
    };

    const card = TestRenderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <CountryCard key={country.id} country={country}/>
        </MemoryRouter>
      </Provider>
    ).toJSON();
  
    expect(card).toMatchSnapshot();
  });
});