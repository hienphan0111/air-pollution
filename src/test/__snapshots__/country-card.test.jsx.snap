// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render the country lists Renders the country correctly 1`] = `
<div
  className="country-card"
>
  <div
    className="overlay"
  />
  <a
    href="/cities"
    onClick={[Function]}
  >
    <div
      className="next-route"
      onClick={[Function]}
    >
      <svg
        className="w-6 h-6 next-icon"
        fill="none"
        stroke="currentColor"
        strokeWidth={1.5}
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12.75 15l3-3m0 0l-3-3m3 3h-7.5M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
          strokeLinecap="round"
          strokeLinejoin="round"
        />
      </svg>
    </div>
  </a>
  <div
    className="flag"
  >
    <img
      alt="Poland-flag"
      src="https://flagcdn.com/pl.svg"
    />
  </div>
  <div
    className="info"
  >
    <h2
      className="name"
    >
      Poland
    </h2>
    <span>
      Asia
    </span>
  </div>
</div>
`;
